File: frontend/src/lib/auth/roles.js
export const ROLES = {
  ADMIN: "admin",
  COACH: "coach",
  STUDENT: "student",
};

const DASHBOARD_PATHS = {
  [ROLES.ADMIN]: "/admin/dashboard",
  [ROLES.COACH]: "/coach/dashboard",
  [ROLES.STUDENT]: "/user/my-programs",
};

/** Returns the post-login dashboard path for a given role. */
export function getDashboardPath(role) {
  return DASHBOARD_PATHS[role] ?? DASHBOARD_PATHS[ROLES.STUDENT];
}

/** Returns where to send the user right after login/register. */
export function getPostLoginPath(role, isProfileComplete = true) {
  if (role === ROLES.STUDENT && !isProfileComplete) {
    return "/user/onboarding";
  }
  return getDashboardPath(role);
}

/** Panel route prefixes guarded by role. */
export const PANEL_PREFIXES = {
  [ROLES.ADMIN]: "/admin",
  [ROLES.COACH]: "/coach",
  [ROLES.STUDENT]: "/user",
};

File: mobile/lib/src/features/auth/application/roles.dart
/// User roles, mirroring `frontend/src/lib/auth/roles.js`.
enum AppRole {
  admin('admin'),
  coach('coach'),
  student('student'),
  unknown('');

  const AppRole(this.value);
  final String value;

  static AppRole fromString(String? raw) {
    return switch (raw) {
      'admin' => AppRole.admin,
      'coach' => AppRole.coach,
      'student' => AppRole.student,
      _ => AppRole.unknown,
    };
  }
}

/// Post-login landing route for a role (student incomplete profile -> onboarding).
String postLoginPath(AppRole role, {bool isProfileComplete = true}) {
  if (role == AppRole.student && !isProfileComplete) return '/onboarding';
  return switch (role) {
    AppRole.coach => '/coach',
    AppRole.student => '/student',
    // Admin isn't a mobile target; send them to the student shell as a fallback.
    _ => '/student',
  };
}
